DirectProgramming/DPC++FPGA/ReferenceDesigns/mvdr_beamforming/src/DiagReciprocal.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef __DIAG_RECIPROCAL__
+#define __DIAG_RECIPROCAL__
+
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+#include "mvdr_complex.hpp"
+
+// SubmitDiagReciprocalKernel
+// Accept an upper-triangular R Matrix from QR Decomposition from StreamingQRD.
+// Compute the reciprocals of the diagonal and write them to an output pipe.
+
+template <typename DiagReciprocalKernelName, // name to use for kernel
+            size_t num_rows,      // Number of rows in R Matrix
+            typename RMatrixInPipe, // The R Matrix input
+            typename RDiagRecipVectorOutPipe // 1 / values on diagonal of R Matrix input
+        >
+event SubmitDiagReciprocalKernel(queue& q) {
+    auto e = q.submit([&](handler& h) {
+        h.single_task<DiagReciprocalKernelName>([=] {
+            int total_reads = num_rows * (num_rows+1) / 2;
+            int row = 1;
+            int col = 1;
+
+            for (int i = 0; i < total_reads; i++) {
```

Comment:
It looks like this kernel will only calculate the reciprocal diagonals for a single matrix.  I believe we'll need to wrap this in a while(true) loop so it will run continuously like the rest of the kernels.